ambry-messageformat/src/test/java/com.github.ambry.messageformat/MessageFormatRecordTest.java

```diff
@@ -151,13 +152,45 @@ public void testMetadataContentRecordV1()
       throws IOException, MessageFormatException {
     // Test Metadata Blob V1
     List<StoreKey> keys = getKeys(60, 5);
-    ByteBuffer metadataContent = getSerializedMetadataContent(keys);
-    List<StoreKey> outKeys = deserializeMetadataContent(metadataContent, new MockIdFactory());
-    Assert.assertEquals("List of keys dont match", keys, outKeys);
+    ByteBuffer metadataContent = getSerializedMetadataContentV1(keys);
+    CompositeBlobInfo compositeBlobInfo = deserializeMetadataContentV1(metadataContent, new MockIdFactory());
+    Assert.assertEquals("Chunk size doesn't match", CompositeBlobInfo.UNDEFINED_CHUNK_SIZE,
+        compositeBlobInfo.getChunkSize());
+    Assert.assertEquals("Total size doesn't match", CompositeBlobInfo.UNDEFINED_TOTAL_SIZE,
+        compositeBlobInfo.getTotalSize());
+    Assert.assertEquals("List of keys dont match", keys, compositeBlobInfo.getKeys());
     // no testing of corruption as we metadata content record doesn't have crc
   }
 
-  private ByteBuffer getSerializedMetadataContent(List<StoreKey> keys) {
+  @Test
+  public void testMetadataContentRecordV2()
+      throws IOException, MessageFormatException {
+    // Test Metadata Blob V2
+    List<StoreKey> keys = getKeys(60, 5);
+    int chunkSize = ThreadLocalRandom.current().nextInt(1, Integer.MAX_VALUE);
+    long totalSize = (long) keys.size() * chunkSize;
```

Comment:
if you don't have one, could you explicitly test the case where `totalSize` is not a multiple of `chunkSize`?